src/extension/index.ts

```diff
@@ -48,6 +52,65 @@ export async function activate(context: ExtensionContext) {
     const panel = new Panel(context, baser, store);
     disposables.push(commands.registerCommand('sarif.showPanel', () => panel.show()));
 
+    // URI handler
+    disposables.push(window.registerUriHandler({
+        async handleUri(uri: vscode.Uri) {
+            if (uri.path === '/alert') {
+                // Launched by Azure DevOps Advanced Security alert page.
+                const pairs = uri.query.split('&');
+                // Find the url parameter.
+                for (const param in pairs) {
+                    const pair = param.split('=');
+                    if (pair[0] === 'url') {
+                        // Decode the alert API URL and pass it to the load function.
+                        await loadAlertSarif(new URL(decodeURIComponent(pair[1])));
+                        break;
+                    }
+                }
+            }
+        }
+    }));
+
+    async function loadAlertSarif(url: URL) {
+        try {
+            const session = await vscode.authentication.getSession('microsoft', ['499b84ac-1321-427f-aa17-267ca6975798/.default'], { createIfNone: true });
+            const accessToken = session?.accessToken;
+
+            const response = await fetch(url, {
```

Comment:
Is there a potential vulnerability here? `url` is coming from user controlled data. So, it could be pointing to any host. This could be a way to steal microsoft access tokens. Should the URL be validated before calling fetch?